### PR TITLE
Schema validation

### PR DESCRIPTION
**Pull request by p-galligan**
Unsure about whether I have to load `transform` as json file first, but I think this is what I've hit upon.

### CODE REVIEW

File: transformer/tests.py [hunk 1]
```diff
@@ -32,6 +34,18 @@
 
 class TransformerTest(TestCase):
 
+    def validate_trasnformers(self):
```
Comment (helrond) [inline transformer/tests.py]:
```suggestion
    def validate_transformers(self):
```

**Comment by p-galligan** [comment]
Combined the two functions as per suggestion.

**Comment by helrond** [comment]
Ok, this is failing now, which is expected. Do you want me to merge this in as-is, or would you prefer to work on getting the transformations to validate first?

**Comment by p-galligan** [comment]
I'll work on transformation validations first, I think.

File: transformer/resources.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ class Subnote(odin.Resource):
 
 class Note(odin.Resource):
     type = odin.StringField(choices=resource_configs.NOTE_TYPE_CHOICES)
-    title = odin.StringField()
+    title = odin.StringField(null=True)
```
Comment (helrond) [inline transformer/resources.py]: Hmm I'm not sure I agree with this - this means that we'll have to do some work either in the indexer or the frontend to infer titles from type. I'd honestly do that work in the transformer.

**Comment by helrond** [comment]
@p-galligan I believe we are good to go on this PR. Can you review once more and make sure everything is working as expected?

**Comment by p-galligan** [comment]
Thank you for taking a look at this! I've tested and it seemingly works for me too. Merging.

**Pull request merged**